src/wincng.c

```
@@ -1679,6 +1795,1185 @@ _libssh2_wincng_dsa_free(libssh2_dsa_ctx *dsa)
+
+    /* Validate parameters */
+    if (!key) {
+        return LIBSSH2_ERROR_INVAL;
+    }
+
+    *key = NULL;
+
+    assert(point->x_len == point->y_len);
+
+    /* Initialize a blob to import */
+    ecc_blob_len = sizeof(BCRYPT_ECCKEY_BLOB) + point->x_len + point->y_len;
+    ecc_blob = malloc(ecc_blob_len);
+    if (!ecc_blob) {
+        return LIBSSH2_ERROR_ALLOC;
+    }
+
+    ecc_blob->dwMagic = _libssh2_wincng_ecdsa_algorithms[point->curve].public_import_magic[keytype];
+    ecc_blob->cbKey = point->x_len;
+
+    /** Copy x, y */
+    memcpy(
+        (PUCHAR)ecc_blob + sizeof(BCRYPT_ECCKEY_BLOB),
+        point->x,
+        point->x_len);
+    memcpy(
+        (PUCHAR)ecc_blob + sizeof(BCRYPT_ECCKEY_BLOB) + point->x_len,
+        point->y,
+        point->y_len);
+
+    status = BCryptImportKeyPair(
+        keytype == WINCNG_ECC_KEYTYPE_ECDSA
+            ? _libssh2_wincng.hAlgECDSA[point->curve]
+            : _libssh2_wincng.hAlgECDH[point->curve],
+        NULL,
+        BCRYPT_ECCPUBLIC_BLOB,
+        key,
+        (PUCHAR)ecc_blob,
+        (ULONG)ecc_blob_len,
+        0);
+    if (!BCRYPT_SUCCESS(status)) {
+        result = LIBSSH2_ERROR_PUBLICKEY_PROTOCOL;
+        goto cleanup;
+    }
+    
+    result = LIBSSH2_ERROR_NONE;
+
+cleanup:
+    free(ecc_blob);
+    return result;
+}
+
+/*
+ * Create a CNG private key from an ECC point.
+ */
+static int
+_libssh2_wincng_privatekey_from_point(IN _libssh2_wincng_ecc_keytype keytype,
+                                      IN _libssh2_ecdsa_point* q,
+                                      IN unsigned char* d,
+                                      IN size_t d_len,
+                                      OUT BCRYPT_KEY_HANDLE* key)
+{
+    int result = LIBSSH2_ERROR_NONE;
+    NTSTATUS status;
+
+    PBCRYPT_ECCKEY_BLOB ecc_blob;
+    size_t ecc_blob_len;
+
+    /* Validate parameters */
+    if (!key) {
+        return LIBSSH2_ERROR_INVAL;
+    }
+
+    *key = NULL;
+
+    assert(q->x_len == q->y_len);
```

May want to return a useful error here instead of dying.